### PR TITLE
fix outofbounds exception when kdoc line is empty

### PR DESCRIPTION
**Pull request by Strum355**
Closes https://github.com/sourcegraph/lsif-kotlin/issues/45

### Test plan

Added unit test

### CODE REVIEW

File: semanticdb-kotlinc/src/main/kotlin/com/sourcegraph/semanticdb_kotlinc/SemanticdbTextDocumentBuilder.kt [hunk 1]
```diff
@@ -133,6 +133,7 @@ class SemanticdbTextDocumentBuilder(
         if (kdoc.isEmpty()) return kdoc
         val out = StringBuilder().append("\n\n").append("----").append("\n")
         kdoc.lineSequence().forEach { line ->
+            if (line.isEmpty()) return ""
```
Comment (keynmol) [inline semanticdb-kotlinc/src/main/kotlin/com/sourcegraph/semanticdb_kotlinc/SemanticdbTextDocumentBuilder.kt:136]: Is this strictly equivalent to the proposed fix? 

```kotlin
            if (line.length <= 0) { continue } // Suggested adding this check
```

This to me reads "skip empty lines, but eventually process every line"

The version in this PR reads "the moment you encounter an empty line anywhere in the doc, terminate the entire function with an empty string"

Unless in Kotlin return + forEach have some different semantics

Comment (Strum355) [reply semanticdb-kotlinc/src/main/kotlin/com/sourcegraph/semanticdb_kotlinc/SemanticdbTextDocumentBuilder.kt:136]: > The version in this PR reads "the moment you encounter an empty line anywhere in the doc, terminate the entire function with an empty string"

Yup youre absolutely right, I messed up the test while I was checking semantics of kdoc comments (line prefixed with `*` or not etc) and forgot actually include the exception case :face_exhaling:

Comment (Strum355) [reply semanticdb-kotlinc/src/main/kotlin/com/sourcegraph/semanticdb_kotlinc/SemanticdbTextDocumentBuilder.kt]: Fixed https://github.com/sourcegraph/lsif-kotlin/pull/54/commits/4d77f55201a735192379cbb1a93192b797c82939

File: semanticdb-kotlinc/src/test/kotlin/com/sourcegraph/semanticdb_kotlinc/test/SemanticdbSymbolsTest.kt [hunk 2]
```diff
@@ -590,4 +592,44 @@ class SemanticdbSymbolsTest {
...
+
+    @TestFactory
+    fun kdoc() =
+        listOf(
+            ExpectedSymbols(
+                "empty kdoc line",
+                SourceFile.testKt(
+                    """
+                    |/**
+                    |
```
Comment (keynmol) [inline semanticdb-kotlinc/src/test/kotlin/com/sourcegraph/semanticdb_kotlinc/test/SemanticdbSymbolsTest.kt:604]: To make the test cover the case I mentioned in my previous comment, how about we put an empty line **amidst** non-empty ones, to ensure that all the lines of kdoc are collected?

Comment (Strum355) [reply semanticdb-kotlinc/src/test/kotlin/com/sourcegraph/semanticdb_kotlinc/test/SemanticdbSymbolsTest.kt:604]: Is there a difference between whats here and your recommendation? Not sure if Im missing what youre thinking here, from what Im getting it seems like this test case should be almost identical in that with the previous code, itd return empty string as it hits the empty line and terminates

**Pull request merged**